### PR TITLE
Add profile model and user model for database storage :issue #7

### PR DESCRIPTION
**Pull request by JAFAJ0**
Two databases, one is "one to one", use userid to identify. Can only have one identical userid.
The other, "many to many".  Can have the same username for many records. but cannot have same date for same person

### What this PR does (required):
Profile model
- userId(required as objectId)
- First Name(required)
- Last Name(required）
- Birthdate(With a default 1998-06-15 date)
- email（required, unique）
- phone number（required）
- address
- Description (Description about the User Profile, will be written by user self)
- photo URL( For future functions,etc)
Availability
- userId(required as objectId)
- start date(should be"YYYY-MM-DD:hh:mm", required, unique)
- end date(should be"YYYY-MM-DD:hh: mm", required, unique)

### Any information needed to test this feature (required):
- These are two pure models without any controller don't believe it could be tested maybe just verify with knowledge.

### CODE REVIEW

File: server/models/Availablity.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcryptjs");
```
Comment (ezeilo-su) [inline server/models/Availablity.js:2]: Can't see where you used this. You can remove it if not used.

Comment (JAFAJ0) [reply server/models/Availablity.js]: Yeah sure forgot to remove them when copy from user model

File: server/models/Profile.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    },
+    lastName: {
+        type: String,
+        required: true,
+    },
+    birthDate: {
+        type: Date,
+        default: '1998-06-15'
+    },
+    email: {
```
Comment (rajivtitus) [inline server/models/Profile.js:21]: Please remove the 'email' field as it is already a part of the User model, we wouldn't want to duplicate data within the DB :)

File: server/models/Profile.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    },
+    address: {
+        type: String,
+        default: ''
+    },
+    description: {
+        type: String,
+        default: ''
+    },
+    photoUrl: {
```
Comment (rajivtitus) [inline server/models/Profile.js:38]: 👍🏽

File: server/models/Availablity.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+const mongoose = require("mongoose");
+
+const availabilitySchema = new mongoose.Schema({
+    userId: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: "User",
+    required: true,
+    },
+    startDate: {
```
Comment (rajivtitus) [inline server/models/Availablity.js:9]: Good start on the availability schema, please explain how you plan to use this design for our project use cases? :)

File: server/models/Profile.js (hunk 2 above)
Comment (JAFAJ0) [reply server/models/Profile.js]: Removed!

File: server/models/Availablity.js (hunk 4 above)
Comment (JAFAJ0) [reply server/models/Availablity.js:9]: The userID is to identify which user contains such available periods.
startDate-endDate identify a period that belongs to the user. 
Its like "many to many" model in relational dabase
Ex:
For the Mocks for available, you can see that listed one by one, which will be store as separate records.

**Comment by JAFAJ0** [comment]
> Great work here! Just a couple of comments, please have a look and address them 😄

Addressed!

File: server/models/Availablity.js (hunk 4 above)
Comment (rajivtitus) [reply server/models/Availablity.js:9]: @JAFAJ0 The mocks have availability for days of the week, 'Monday Tuesday...and so on'. How will we implement such a use case with this design? 😄

Comment (JAFAJ0) [reply server/models/Availablity.js]: My idea is to store multiple available periods and assume we want to store more than one week's data. 
With the start/end date time, for instance, from 15/08/2016 10 am-18/09/2016 3:15 pm
The front end could know which day is it in a week, or I modify the get controller later to let the front end knows with a certain date.
In my opinion, when the front end fetches a specified period, the backend gathers information, sorts them, and returns it available in that week.
Your model is clear in which day is in a week, but I think that is quite ambiguous in date and time.

**Comment by JAFAJ0** [comment]
> @JAFAJ0 please readjust the base of the PR, the files pertaining to this PR aren't visible.

Changed.

File: server/models/Availablity.js (hunk 4 above)
Comment (rajivtitus) [reply server/models/Availablity.js:9]: @JAFAJ0 thanks for the clarification! 👍🏽

**Pull request merged**